examples/gateway/02-gateway.yaml

```diff
@@ -0,0 +1,16 @@
+---
+kind: Gateway
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: contour
+  namespace: projectcontour
+spec:
+  gatewayClassName: example
```

Comment:
`gatewayClassName` is a reference to a `gatewayclass.metadata.name`. The current example gatewayclass is named "example".  The `spec.controller` of the referenced gatewayclass must match what contour expects as described by the contour config `gateway.controllerName` parameter (currently `projectcontour.io/projectcontour/contour`).

Reply:
Ohh it is ClassName, I was thinking all along that was ControllerName. Why then do we care about the controller name field? Can we just reference the class by name?

Reply:
We care about the controller field so Contour knows which gatewayclasses to process. If a gateway references a gatewayclass by name, the gateway controller will get that gatewayclass and check that the controller field matches what's configured. If it does not match the configured controllerName, then the gateway should be considered invalid.

Reply:
We just need to understand if Contour will ever process more than one Gateway described in https://github.com/projectcontour/contour/issues/3825. 

If not, then to me it makes more sense to only require the ClassName to be configured in the config file since now we have a specific class to match against.

Reply:
> We just need to understand if Contour will ever process more than one Gateway described in #3825.

This is a must-have to support knative's use case. I see multi-gateway support as one of the next big follow-on's to this PR.